File: app/src/main/java/com/system/m4/views/home/HomeContract.java
package com.system.m4.views.home;

import android.content.Context;
import android.support.annotation.NonNull;

import com.system.m4.views.vos.RedirectButtomVO;
import com.system.m4.views.vos.TagSummaryVO;
import com.system.m4.views.vos.TransactionVO;
import com.system.m4.views.vos.VOItemListInterface;

import java.util.List;

/**
 * Created by eferraz on 29/04/17.
 * For M4
 */

public class HomeContract {

    /**
     *
     */
    interface View {

        void setPresenter(Presenter presenter);

        void setListTransactions(List<VOItemListInterface> listVo);

        void showTransactionDialog(TransactionVO vo);

        void showError(String message);

        void showSuccessMessage(int template, int param);

        void openDeleteDialog(TransactionVO item);

        void requestShowListTransaction(TagSummaryVO item);

        Context getContext();

        void openDetail(RedirectButtomVO vo);

        void showPoupu(android.view.View viewClicked, TransactionVO vo);
    }

    /**
     *
     */
    public interface Presenter {

        void init(int relativePosition, int homeVisibility);

        void requestListTransaction();

        void selectItem(TransactionVO vo);

        void requestCopy(@NonNull TransactionVO item);

        void requestDelete(@NonNull TransactionVO item);

        void onConfirmDeleteClicked(TransactionVO item);

        void requestShowListTransaction(@NonNull TagSummaryVO item);

        void onClickVO(VOItemListInterface vo);

        boolean onLongClickVO(VOItemListInterface vo, android.view.View view);
    }
}

File: app/src/main/java/com/system/m4/kotlin/contracts/HomeContract.kt
package com.system.m4.kotlin.contracts

import android.content.Context

import com.system.m4.labs.vos.TagSummaryVO
import com.system.m4.labs.vos.TransactionVO
import com.system.m4.labs.vos.VOItemListInterface

/**
 * Created by eferraz on 29/04/17.
 * For M4
 */

interface HomeContract {

    /**
     *
     */
    interface View {

        fun getContext(): Context

        fun setPresenter(presenter: Presenter)

        fun setListTransactions(listVo: List<VOItemListInterface>)

        fun showTransactionDialog(vo: TransactionVO)

        fun showError(message: String)

        fun showSuccessMessage(template: Int, param: Int)

        fun openDeleteDialog(item: TransactionVO)

        fun showListTransaction(item: TagSummaryVO)

        fun showPoupu(viewClicked: android.view.View, vo: TransactionVO)
    }

    /**
     *
     */
    interface Presenter {

        fun init(relativePosition: Int)

        fun requestListTransaction()

        fun selectItem(vo: TransactionVO)

        fun requestCopy(item: TransactionVO)

        fun requestDelete(item: TransactionVO)

        fun onConfirmDeleteClicked(item: TransactionVO)

        fun requestShowListTransaction(item: TagSummaryVO)

        fun onClickVO(vo: VOItemListInterface)

        fun onLongClickVO(vo: VOItemListInterface, view: android.view.View): Boolean
    }
}
